src/providers/postgres/raster/qgspostgresrasterprovider.h

```diff
@@ -0,0 +1,207 @@
+
+    // QgsRasterInterface interface
+    virtual Qgis::DataType dataType( int bandNo ) const override;
+    virtual int bandCount() const override;
+    virtual QgsRasterInterface *clone() const override;
+    virtual Qgis::DataType sourceDataType( int bandNo ) const override;
+    virtual int xBlockSize() const override;
+    virtual int yBlockSize() const override;
+    virtual QgsRasterBandStats bandStatistics( int bandNo, int stats, const QgsRectangle &extent, int sampleSize, QgsRasterBlockFeedback *feedback ) override;
+
+    // QgsRasterDataProvider interface
+    virtual QString htmlMetadata() override;
+    virtual QString lastErrorTitle() override;
+    virtual QString lastError() override;
+    int capabilities() const override;
+
+    // QgsRasterInterface interface
+    int xSize() const override;
+    int ySize() const override;
+
+    static const QString PG_RASTER_PROVIDER_KEY;
+    static const QString PG_RASTER_PROVIDER_DESCRIPTION;
+
+  private:
+
+    bool mValid = false;
+    QgsCoordinateReferenceSystem mCrs;
+    //! Data source URI struct for this layer
+    QgsDataSourceUri mUri;
+    //! provider references query (instead of a table)
+    bool mIsQuery;
+    //! Name of the table with no schema
+    QString mTableName;
+    //! Name of the table or subquery
+    QString mQuery;
+    //! Name of the raster column
+    QString mRasterColumn;
+    //! Name of the schema
+    QString mSchemaName;
+    //! SQL statement used to limit the features retrieved
+    QString mSqlWhereClause;
+    //! Rectangle that contains the extent (bounding box) of the layer
+    mutable QgsRectangle mExtent;
+    //! Use estimated metadata. Uses fast table counts, geometry type and extent determination
+    bool mUseEstimatedMetadata = true;
+    //! Error information
+    QString mError;
+    //! Error title
+    QString mErrorTitle;
+    //! Data type for each band
+    std::vector<Qgis::DataType> mDataTypes;
+    //! Data size in bytes for each band
+    std::vector<int> mDataSizes;
+    //! Store overviews
+    QMap<unsigned int, QString> mOverViews;
+    //! Band count
+    int mBandCount = 0;
+    //! If is tiled
+    bool mIsTiled = false;
+    //! If is out of DB
+    bool mIsOutOfDb = false;
+    //! Has spatial index
+    bool mHasSpatialIndex = false;
+    //! Raster size x
+    long mWidth = 0;
+    //! Raster size y
+    long mHeight = 0;
+    //! Raster tile size x
+    int mTileWidth = 0;
+    //! Raster tile size y
+    int mTileHeight = 0;
+    //! Scale x
+    double mScaleX = 0;
+    //! Scale y
+    double mScaleY = 0;
+
+    QString mDetectedSrid;            //!< Spatial reference detected in the database
+    QString mRequestedSrid;           //!< Spatial reference requested in the uri
+    QgsPostgresConn *mConnectionRO = nullptr ; //!< Read-only database connection (initially)
+    QgsPostgresConn *mConnectionRW = nullptr ; //!< Read-write database connection (on update)
```

Comment:
Do we need this? It's all read-only, right?

Reply:
For now, yes. But I think editing will be possible in the future, I'd keep it even if it's currently unused.